FindGridSpots: added transformation method selection.

tepals opened the pull request:
- Sometimes it is necessary that the returned grid is a perfect grid, with equal scaling in x and y and no shear. With the possibility to select the similarity method this is now possible.
- Refactored the transformation calls to not use the deprecated call method, but use the apply method.

Sabrina85 commented on src/odemis/acq/align/test/spot_test.py:
```diff
@@ -211,6 +211,33 @@ def test_find_grid(self):
+        image[54:150:12, 54:150:12] = 1
+        spot_coordinates, translation, scaling, rotation, shear = spot.FindGridSpots(image, (8, 8), method='similarity')
+        self.assertAlmostEqual(rotation, 0, places=4)
+        self.assertEqual(shear, None)
```
There is an assertIsNone function that you can use :)

Sabrina85 reviewed (changes requested):
I have only two small comments on the tests. The rest looks fine!

pieleric commented on src/odemis/acq/align/spot.py:
```diff
@@ -354,6 +354,11 @@ def FindGridSpots(image, repetition):
         Data array containing the greyscale image.
     repetition : tuple of ints
         Number of expected spots in (X, Y).
+    method : str, "affine" or "similarity"
```
Instead of a "str", please create two constants (like `GRID_AFFINE`, `GRID_SIMILARITY`) and use them here.
This way, it's clearer that a it's not *any* string, and also you'll reduce chances of typos in the name.

pieleric commented on src/odemis/acq/align/test/spot_test.py:
```diff
@@ -211,6 +211,33 @@ def test_find_grid(self):
+        image = numpy.zeros((256, 256))
+        # set a grid of 8 by 8 points to 1
+        image[54:150:12, 54:150:12] = 1
+        spot_coordinates, translation, scaling, rotation, shear = spot.FindGridSpots(image, (8, 8), method='similarity')
```
So the two methods return... exactly the same thing (excepted that shear = None)? That feels a little weak as test case. Could you find a different input where the output is really different between the 2 methods?

tepals commented:
I added another test case. In the process of creating, I found out that the minimum amount of spots in the grid is 6, therefore I added checks for that as well.
